easybuild/easyblocks/g/gromacs.py

```diff
@@ -493,3 +562,119 @@ def sanity_check_step(self):
             'dirs': dirs,
         }
         super(EB_GROMACS, self).sanity_check_step(custom_paths=custom_paths)
+
+    def run_all_steps(self, *args, **kwargs):
+        """
+        Put configure options in place for different variants, (no)mpi, single/double precision.
+        """
+        # Save installopts so we can reset it later. The gmxapi pip install
+        # can't handle the -j argument.
+        self.save_installopts = self.cfg.get('installopts')
+
+        # keep track of config/build/installopts specified in easyconfig
+        # file, so we can include them in each iteration later
+        common_config_opts = self.cfg.get('configopts')
+        common_build_opts = self.cfg.get('buildopts')
+        common_install_opts = self.cfg.get('installopts')
+
+        self.save_install_cmd = self.cfg.get('install_cmd')
+        self.save_build_cmd = self.cfg.get('build_cmd')
+
+        self.cfg['configopts'] = []
+        self.cfg['buildopts'] = []
+        self.cfg['installopts'] = []
+
+        if LooseVersion(self.version) < LooseVersion('4.6'):
+            prec_opts = {
+                'single': '--disable-double',
+                'double': '--enable-double',
+            }
+            mpi_type_opts = {
+                'nompi': '--disable-mpi',
+                'mpi': '--enable-mpi'
+            }
+            # Double precision pattern so search for in configopts
+            self.DP_pattern = '--enable-double'
```

Comment:
nitpick: Use that in `prec_opts`?

Reply:
Did it the other way around instead since that makes more sense.